poms/tycho/pom.xml

```diff
@@ -214,11 +214,31 @@
 
 	<repositories>
 
-        <!-- SmartHome p2 repository -->
+        <!-- SmartHome p2 snapshot repository -->
         <repository>
-            <id>p2-smarthome</id>
+            <id>p2-smarthome-snapshot</id>
             <url>https://openhab.jfrog.io/openhab/eclipse-smarthome-stable</url>
             <layout>p2</layout>
+            <releases>
+                <enabled>false</enabled>
+            </releases>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+        </repository>
+
+        <!-- SmartHome p2 release repository -->
+        <repository>
+            <id>p2-smarthome-release</id>
+            <url>http://download.eclipse.org/smarthome/updates-release/${esh.version}</url>
```

Comment:
Hm, this causes now an issue on the pipeline:
```
[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-smarthome-release' from location http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT: No repository found at http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT. -> [Help 1]
```
@pfink Any clever idea how to handle this best? It is correct that this repo does not exist for esh snapshots, so IF esh.version contains "-SNAPSHOT", we must not use this repository.

Reply:
Why is this repo then enabled for snapshots? https://github.com/openhab/openhab-core/blob/master/poms/tycho/pom.xml#L240

Reply:
It is enabled for snapshot builds and this is what we are doing here: An openHAB 2.2.0-SNAPSHOT build, which uses an ESH release version, so we need access to this repo.

Reply:
I think you misunderstood this configuration option: It does refer to the artifact that should be downloaded and not to the artifact that is currently built.

Reply:
For a test I did 
```
        <!-- SmartHome p2 release repository -->
        <repository>
            <id>p2-smarthome-release</id>
            <url>http://download.eclipse.org/smarthome/updates-release/${esh.version}</url>
            <layout>p2</layout>
            <releases>
                <enabled>true</enabled>
                <updatePolicy>never</updatePolicy>
            </releases>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
        </repository>
```
and set esh.version=0.10.0-SNAPSHOT.
The situation isn't any better:
```
[INFO] Adding repository https://openhab.jfrog.io/openhab/eclipse-smarthome-stable
[INFO] Adding repository http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT
[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-smarthome-release' from location http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT: No repository found at http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT. -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-smarthome-release' from location http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT
```

How can we solve this best? Shall we manually remove this repo from the pom, whenever we refer to esh snapshots and add it again, if we switch to a release? But that would need to be done in the pipeline as well (as this is where it currently breaks).

Reply:
Mhm, it seems like it still wants to resolve the repository url even though all artifacts can be grabbed from other repositories... Two possible solutions I see:
* Do not use esh.version as a property, use a separate property (e.g. `esh.releaseVersion`)
* Put it into a profile that is only used during the release build like here: https://github.com/openhab/openhab-distro/blob/master/pom.xml#L215

Reply:
Which solution would you prefer?

Reply:
They're both not perfect, but I'd like the first one more as it does not require passing extra arguments to the release build to make it work.

Reply:
Btw: Even though it was not the final solution, this repo should still be disabled for snapshots.